### DISCUSSION TITLE
Re: Any reason using pointers in service methods

### DISCUSSION THREAD

**Discussion by huyfm** [benbjohnson/wtf#88] [Q&A] [answered]
I have read a lot of articles recommended to use values in Go to improve cache locality. So is there any practical reason behind using pointers across all services functions in wtf ?
For example,
```
type DialService interface {
	FindDialByID(ctx context.Context, id int) (*Dial, error)
	FindDials(ctx context.Context, filter DialFilter) ([]*Dial, int, error)
	CreateDial(ctx context.Context, dial *Dial) error
	UpdateDial(ctx context.Context, id int, upd DialUpdate) (*Dial, error)
	DeleteDial(ctx context.Context, id int) error
}
```

**Comment by benbjohnson** [comment, marked as answer]
You could also do non-pointers but I find that using pointers tends to make the calling code a bit easier to use. It's rare that pointer versus non-pointer code has a noticeable impact unless you're doing tight loops or a large volume of data.
